src/coreclr/jit/lowerxarch.cpp

```diff
@@ -795,15 +795,15 @@ void Lowering::LowerCast(GenTree* tree)
     //   srcType = float/double                    castToType = * and overflow detecting cast
     //       Reason: must be converted to a helper call
     //   srcType = float/double,                   castToType = ulong
-    //       Reason: must be converted to a helper call
+    //       Reason: must be converted to a helper call unless we have AVX512F
     //   srcType = uint                            castToType = float/double
     //       Reason: uint -> float/double = uint -> long -> float/double
     //   srcType = ulong                           castToType = float
     //       Reason: ulong -> float = ulong -> double -> float
-    if (varTypeIsFloating(srcType))
+    if (srcType == TYP_FLOAT)
```

Comment:
Is it problematic that this isn't including double anymore? The comments above seem to indicate that can hit this, in which case I'd expect a comment elaborating why its not relevant to handle

Reply:
I am changing it back to original code. This code would not be hit because there is already a check in morph.cpp to not use the ```JIT_DBL2Ulng``` function only if AVX512F ISA is present. Hence, moving to original code as was to help debug.